src/models/PrimaryAuth.js

```diff
@@ -141,6 +161,13 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
       if (!this.settings.get('features.passwordlessAuth')) {
         signInArgs.password = this.get('password');
       }
+
+      // if its an idx stateToken, we send the parameter as identifier to login API
+      if (this.appState.get('isIdxStateToken')) {
+        signInArgs.identifier = signInArgs.username;
+        delete signInArgs.username;
+        delete signInArgs.options;
```

Comment:
maybe a comment on why we delete these options as well

Reply:
yeah... or maybe there should be a main if/else block to add whichever data we need depending on `isIdxStateToken`, instead of deleting options we aded a few lines above